core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -2791,17 +2797,17 @@ private List<Type> analyzeJsonPathInvocation(String functionName, Expression nod
                     else if (isNumericType(parameterType) || parameterType.equals(BOOLEAN)) {
                         passedType = parameterType;
                     }
-                    else if (isDateTimeType(parameterType)) {
-                        if (parameterType.equals(INTERVAL_DAY_TIME) || parameterType.equals(INTERVAL_YEAR_MONTH)) {
-                            throw semanticException(INVALID_FUNCTION_ARGUMENT, parameter, "Invalid type of JSON path parameter: %s", parameterType.getDisplayName());
-                        }
+                    else if (isDateTimeType(parameterType) && !parameterType.equals(INTERVAL_DAY_TIME) && !parameterType.equals(INTERVAL_YEAR_MONTH)) {
                         passedType = parameterType;
                     }
                     else {
-                        if (!typeCoercion.canCoerce(parameterType, VARCHAR)) {
-                            throw semanticException(INVALID_FUNCTION_ARGUMENT, parameter, "Invalid type of JSON path parameter: %s", parameterType.getDisplayName());
+                        try {
+                            plannerContext.getMetadata().getCoercion(session, parameterType, VARCHAR);
+                        }
+                        catch (OperatorNotFoundException e) {
+                            throw semanticException(NOT_SUPPORTED, node, "Unsupported type of JSON path parameter: %s", parameterType.getDisplayName());
                         }
-                        coerceType(parameter, parameterType, VARCHAR, "JSON path parameter");
+                        jsonTextArguments.add(NodeRef.of(parameter));
```

Comment:
Why do we need to track these explicitly? It should be sufficient to just register a coercion via `addOrReplaceExpressionsCoercion`.

Reply:
I considered that approach, but so far that mechanism was only used for implicit coercions. I didn't want to mix in the JSON path parameter coercions, which are part of JSON functions semantics rather than SQL type alignment.

Reply:
That mechanism is for anything where the analyzer decides a cast needs to be inserted, so it should be safe to use here. The concept of whether an implicit coercion is allowed is an analysis time concern and the planner doesn't care about *why* a cast needs to be inserted.

Reply:
In fact, we could say that this *is* an implicit coercion, just in a very narrow scope (json function parameters)

Reply:
I switched to the existing mechanism (`addOrReplaceExpressionsCoercion`).